src/Language/PureScript/Bridge/Printer.hs

```diff
@@ -71,10 +71,24 @@ importLineToText l = "import " <> importModule l <> " (" <> typeList <> ")"
     typeList = T.intercalate ", " (Set.toList (importTypes l))
 
 sumTypeToText :: SumType 'PureScript -> Text
-sumTypeToText (SumType t cs) = T.unlines $
+sumTypeToText st@(SumType t cs) = T.unlines $
     "data " <> typeInfoToText True t <> " ="
   : "    " <> T.intercalate "\n  | " (map (constructorToText 4) cs)
-  : [ "\nderive instance generic" <> _typeName t <> " :: Generic " <> _typeName t ]
+  : [ "\nderive instance generic" <> _typeName t <> " :: " <> genericConstrains <> genericInstance t ]
+  where
+    genericInstance = ("Generic " <>) . typeInfoToText False
+    genericConstrains
+        | stpLength == 0 = mempty
+        | otherwise = (<> " => ") $
+            if stpLength == 1
+                then genericConstrainsInner
+                else bracketWrap genericConstrainsInner
+    genericConstrainsInner = T.intercalate ", " $ map genericInstance sumTypeParameters
+    stpLength = length sumTypeParameters
+    bracketWrap x = "(" <> x <> ")"
+    sumTypeParameters = filter isTypeParameter $ getUsedTypes st
+    -- TODO: I find this stupid. Would much rather like some other way to make this check?
+    isTypeParameter tp = T.toLower (_typeName tp) == _typeName tp
```

Comment:
which is t ^. typeParameters, something like:

```haskell
sumTypeParameters = filter (`elem` getUsedTypes st) $ t ^. typeParameters
```

I just noticed from the definition of `getUsedTypes` that it preserves duplicates - so filtering `t ^. typeParameters` is correct, the other way round could result in instances like:

```purescript
instance genericFoo :: (Generic a, Generic a, Generic a) => Generic (Foo a)
```
Obviously it might be bad that getUsedTypes does not filter duplicates - I am going to check, at least it should be documented.